platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/UnionFileCollection.java

```diff
@@ -81,4 +84,35 @@ protected void visitChildren(Consumer<FileCollectionInternal> visitor) {
             visitor.accept(fileCollection);
         }
     }
+
+    @Override
+    public Optional<FileCollectionExecutionTimeValue> calculateExecutionTimeValue() {
+        Set<FileCollectionExecutionTimeValue> sourceExecutionTimeValues = new HashSet<>();
```

Comment:
❌ Use an `ImmutableSet.Builder` for efficiency and correctness (order preserving)